pkg/apis/core/v1alpha1/federatedconfig_types.go

```diff
@@ -29,12 +31,35 @@ import (
 type FederatedConfigSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "kubebuilder generate" to regenerate code after modifying this file
+
+	LimitedScope       bool                 `json:"limited-scope"`
+	RegistryNamespace  string               `json:"registry-namespace,omitempty"`
+	ControllerDuration DurationConfig       `json:"controller-duration"`
+	LeaderElect        LeaderElectConfig    `json:"leader-elect"`
+	FeatureGates       []FeatureGatesConfig `json:"feature-gates"`
```

Comment:
Many of the fields does not have omitempty json tag, although they are optional and can have default setting.